packages/components/TopMenu/NotificationDrawer.tsx

```diff
@@ -0,0 +1,370 @@
+        >
+          <UserNameInline
+            userId={item.triggerBy || ""}
+            showText={false}
+            size="S"
+            style={{ width: "100%" }}
+          />
+        </View>
+
+        <View style={{ flex: 4, marginHorizontal: layout.spacing_x1 }}>
+          <OmniLink
+            to={{
+              screen: "FeedPostView",
+              params: { id: item.action },
+            }}
+          >
+            <View>
+              <BrandText style={fontSemibold14}>
+                {useBuildBodyText(item)}
+              </BrandText>
+              <View
+                style={{
+                  marginTop: layout.spacing_x1,
+                  flexDirection: "row",
+                  flexWrap: "nowrap",
+                  justifyContent: "space-between",
+                }}
+              >
+                <BrandText style={fontSemibold12}>
+                  {moment.unix(item.createdAt).fromNow()}
+                </BrandText>
+                <BrandText
+                  style={[
+                    fontSemibold12,
+                    {
+                      textTransform: "capitalize",
+                    },
+                  ]}
+                >
+                  {item.category}
+                </BrandText>
+              </View>
+            </View>
+          </OmniLink>
+        </View>
+
+        <View
+          style={{
+            flex: 0.5,
+            zIndex: 2,
+            justifyContent: "flex-end",
+            alignSelf: "center",
+          }}
+        >
+          <DismissNotificationButton
+            id={item.id}
+            onPress={() => setVisible(false)}
+          />
+        </View>
+      </View>
+    </View>
+  );
+};
+const NoNotifications: React.FC = () => (
+  <View
+    style={{
+      alignSelf: "center",
+      padding: layout.spacing_x3,
+    }}
+  >
+    <BrandText>No new Notifications 🥺</BrandText>
+  </View>
+);
+
+const Controls: React.FC = () => {
+  const selectedWallet = useSelectedWallet();
+  const queryClient = useQueryClient();
+
+  return (
+    <View
+      style={{
+        padding: layout.spacing_x1,
+        borderBottomWidth: 1,
+        width: "100%",
+        borderBottomColor: neutral44,
+      }}
+    >
+      <TouchableOpacity
+        style={{
+          alignSelf: "flex-end",
+        }}
+        onPress={async () => {
+          await clearAllNotifications(queryClient, {
+            userId: selectedWallet?.userId,
+          });
+        }}
+      >
+        <BrandText style={fontSemibold14}>Clear All</BrandText>
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+const useBuildBodyText = (item: Notification) => {
+  const userInfo = useNSUserInfo(item.triggerBy);
+  const name =
+    userInfo?.metadata?.tokenId || tinyAddress(item.triggerBy, 30) || "";
+
+  if (item.category === "reaction") {
+    return `${name} ${item.body} your post.`;
+  }
+  if (item.category === "comment") {
+    return `${name} commented your post.`;
```

Comment:
I purpose somthing like this (address used in tinyAddress instead of userId + linebreak)
```suggestion
    const [, address] = parseUserId(item.triggerBy)
  const name =
    userInfo?.metadata?.tokenId || `${tinyAddress(address, 24)}` || "";

  if (item.category === "reaction") {
    return `${name}\n${item.body} your post.`;
  }
  if (item.category === "comment") {
    return `${name}\ncommented your post.`;
```